### PR TITLE
For #5318: Close response stream to allow another request.

### PR DESCRIPTION
**Pull request by mcarare**
This also needs UX confirmation in order to land.

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt [hunk 1]
```diff
@@ -206,7 +206,11 @@ class ManualAddSearchEngineSettingsFragment : BaseSettingsFragment() {
             )
 
             return try {
-                client.fetch(request).status < VALID_RESPONSE_CODE_UPPER_BOUND
+                val response = client.fetch(request)
+                // Close the response stream to ensure the body is closed correctly. See https://bugzilla.mozilla.org/show_bug.cgi?id=1603114.
+                response.body.close()
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt:211]: Why does this not happen in Fenix as well? 🤔

Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt:211]: It did. In Fenix, we use `response.body.string()` to read the stream (that also automatically closes it. ) At the moment of the fix the `close()` method had a bug that has been fixed now.  We should also use close there too. I will put up a PR for that.

Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt:211]: https://github.com/mozilla-mobile/fenix/pull/21203

**Comment by mcarare** [comment]
UX also gave the go-ahead: https://github.com/mozilla-mobile/focus-android/issues/5318#issuecomment-915724307

**Review by jonalmeida — APPROVED** [review]
Brilliant, thanks for following up so quickly!

**Pull request merged**